backend/inspirehep/records/marshmallow/literature/ui.py

```diff
@@ -134,12 +135,16 @@ def get_arxiv_fulltext_link(self, data):
             }
         return missing
 
-    def get_internal_fulltext_link(self, data):
+    def get_internal_fulltext_link(self, document):
         if not current_app.config.get("FEATURE_FLAG_ENABLE_FILES"):
             return missing
-        description = data.get("description") or "fulltext"
-        url = data.get("url")
-        if url and not data.get("hidden", False):
+        description = document.get("description") or "fulltext"
+        url = document.get("url")
+        if (
+            url
+            and not self.is_document_hidden(document)
+            and not self.is_document_arxiv(document)
```

Comment:
I'm not sure this check is enough, do we keep ``source`` in the ``documents``?

Reply:
We should keep source in documents, but it's not required field. However this is how we were checking if doc is from arxiv in dojson if I understand this correctly 
https://github.com/inspirehep/inspire-dojson/commit/dec6e94df73165b06a5b15796c64fba81c408803#diff-d3b380993699eaa3534ec6fac751e842

https://github.com/inspirehep/inspire-dojson/blob/dec6e94df73165b06a5b15796c64fba81c408803/inspire_dojson/hep/rules/bdFFT.py#L67

And source is also populated from _get_source() https://github.com/inspirehep/inspire-dojson/blob/dec6e94df73165b06a5b15796c64fba81c408803/inspire_dojson/hep/rules/bdFFT.py#L99

Reply:
I don't see we keep such informaton https://github.com/inspirehep/inspirehep/blob/master/backend/inspirehep/records/api/literature.py#L374 though, do I miss something?

Reply:
Ok, When I'm analyzing this we are not able to figure out if document is from arxiv or not. Previously we were not downloading arxiv documents so all of them had source preserved, but when we download them we forget the source. I will add proper changes to store the source as this is the only way to find out if file is arxiv after downloading it. Otherwise We can just skip downloading of ArXiv files. We are missing stuff in refinement...